Feat/project

JohnsonMao opened the pull request:
1. 專案管理功能優化
- 重新設計 Project API schema，提升了程式碼的可維護性
- 新增專案創建頁面，完善了專案管理流程
- 優化 API 請求處理邏輯，提升了程式碼的可讀性

2. 設定檔結構優化
- 將 config.ts 重命名為 marathon.ts，使命名更符合實際用途
- 將 dayjs 相關設定集中到 _app.jsx，統一管理時間格式化設定

3. CI/CD 流程優化
- 修復 GitHub Actions 工作流程中的死鎖問題

JohnsonMao committed:
fix: resolve workflow deadlock in GitHub Actions

JohnsonMao committed:
feat(project): add create project page and refactor API schema

JohnsonMao committed:
refactor: centralize dayjs config in _app

JohnsonMao committed:
refactor(config): rename config to marathonConfig

JohnsonMao committed:
refactor: useProject hook with SWR mutation

JohnsonMao committed:
refactor: project creation with max limit and input handling

JohnsonMao committed:
feat: add project management features and UI updates

JohnsonMao committed:
style: format code in milestones page

JohnsonMao committed:
feat: add date range picker component

JohnsonMao committed:
refactor: restructure milestone module

- Rename project_id to projectId for consistent naming convention
- Split project services into separate files for better maintainability
- Add useProjectMilestone and useProjectMilestoneList custom hooks
- Enhance milestone page with sorting and filtering features
- Improve type definitions with Zod schema
- Add DateRangePicker for non-marathon projects
- Optimize milestone list rendering performance

JohnsonMao committed:
feat: add milestone creation form

vincentxuu commented on components/Milestones/MilestoneItem.tsx:
```diff
@@ -21,7 +21,7 @@ const MilestoneItem = ({
 
   return (
-    <div className="p-[10px] bg-basic-100 flex flex-col gap-2">
+    <div className="p-2.5 bg-basic-100 flex flex-col gap-2">
```
好奇tailwindcss是不是盡量不寫固定px?這原本是我寫的ＸＤ

JohnsonMao replied:
寫死 px 如果使用者有調整螢幕大小，px 就不會等比例縮放，導致畫面會顯得擁擠

noracami replied:
咦 應該一樣吧 看解出來的 CSS 是一樣的

JohnsonMao replied:
<img width="390" alt="image" src="https://github.com/user-attachments/assets/a365fe32-75aa-4b45-ae80-c61b86170c43" />

我看 p-2.5 是 `padding: 0.625rem /* 10px */;`

JohnsonMao replied:
<img width="450" alt="image" src="https://github.com/user-attachments/assets/dfa63e17-896e-45ed-b37c-430b01ff62cb" />

實際打包也是 `0.625rem`

noracami replied:
哦哦
Rem 會參考最外面那個元素的大小嗎

JohnsonMao replied:
會參考 body 的 `font-size`，某些設備的字體縮放也只是調整 body 的 `font-size`

JohnsonMao committed:
refactor: simplify project creation page SEO

vincentxuu commented on contexts/Project.tsx:
```diff
@@ -43,36 +42,6 @@ export function ProjectProvider({ children }: ProjectContextProviderProps) {
       setIsFetching(false);
     }
   };
-  const fetchMyProject = async () => {
```
這邊重構不知道ruby那邊有沒有影響

JohnsonMao replied:
現況專案沒看到有任何地方有使用 `fetchMyProject` 這個函式，新的 PR 也未看到，所以就先移除了

vincentxuu commented on pages/_app.jsx:
```diff
@@ -223,13 +229,15 @@ const App = ({ Component, pageProps }) => {
-              </AuthProvider>
-            </SnackbarProvider>
-          </SWRConfig>
+          <LocalizationProvider dateAdapter={AdapterDayjs} adapterLocale="zh-tw">
```
讚喔，這是準備I18N嗎？

JohnsonMao replied:
沒耶～只是單純我要顯示中文

vincentxuu commented on services/project/notes.ts:
```diff
@@ -1,8 +1,9 @@
 import { uploadImage } from '../images';
 
+const projectEndpoint = '/projects';
+
```
後端
description改成content
刪除imgUrl 字串
新增imageUrls 陣列
新增videoUrls 陣列

JohnsonMao replied:
喔～對耶！忘了這個

JohnsonMao replied:
已調整

vincentxuu commented on services/project/outcomes.ts:
```diff
@@ -1,8 +1,9 @@
-import { projectEndpoint } from './index';
 import { uploadImage } from '../images';
 
+const projectEndpoint = '/projects';
```
後端
description改成content
刪除imgUrl 字串
新增imageUrls 陣列
新增videoUrls 陣列

JohnsonMao replied:
喔～對耶！忘了這個

JohnsonMao replied:
已調整

vincentxuu commented on shared/components/Collapse.tsx:
```diff
@@ -57,16 +58,32 @@ interface CollapseListProps {
         '*:grid *:transition-[grid-template-rows]',
-        '*:grid-rows-[1fr] *:aria-hidden:grid-rows-[0fr] [&>*>*]:aria-hidden:overflow-hidden',
+        '*:grid-rows-[1fr] *:aria-hidden:grid-rows-[0fr]',
+        !isOpened && '[&>*>*]:overflow-hidden',
```
這邊的用法實在是太酷了，前端好難

JohnsonMao committed:
chore(ci): optimize CI/CD workflow triggers

- Add "**" pattern to include all branches first
- Then exclude deployment branches (prod/dev/feat/v2)
- Avoid duplicate CI checks between CI and CD workflows

JohnsonMao committed:
refactor: support multiple media in notes & outcomes

JohnsonMao committed:
refactor: CI workflow for PR checks

vincentxuu commented:
MIMG
